cmd/podman/common/completion.go

```diff
@@ -23,90 +23,73 @@ var (
 	LogLevels = []string{"debug", "info", "warn", "error", "fatal", "panic"}
 )
 
-func getContainers(toComplete string, statuses ...string) ([]string, cobra.ShellCompDirective) {
+type completeType int
+
+const (
+	// complete names and IDs
+	completeDefault completeType = iota
+	// only complete IDs
+	completeIDs
+	// only complete Names
+	completeNames
+)
+
+func getContainers(toComplete string, cType completeType, statuses ...string) ([]string, cobra.ShellCompDirective) {
 	suggestions := []string{}
 	listOpts := entities.ContainerListOptions{
 		Filters: make(map[string][]string),
 	}
 	listOpts.All = true
 	listOpts.Pod = true
+	if len(statuses) > 0 {
+		listOpts.Filters["status"] = statuses
+	}
 
-	// TODO: The api doesn't handle several different statuses correct see:
-	// https://github.com/containers/podman/issues/8344
-	// Instead of looping over the statuses we should be able to set
-	// listOpts.Filters["status"] = statuses
-
-	var containers []entities.ListContainer
-	var err error
-	if len(statuses) == 0 {
-		containers, err = registry.ContainerEngine().ContainerList(registry.GetContext(), listOpts)
-		if err != nil {
-			cobra.CompErrorln(err.Error())
-			return nil, cobra.ShellCompDirectiveError
-		}
-	} else {
-		for _, s := range statuses {
-			listOpts.Filters["status"] = []string{s}
-			res, err := registry.ContainerEngine().ContainerList(registry.GetContext(), listOpts)
-			if err != nil {
-				cobra.CompErrorln(err.Error())
-				return nil, cobra.ShellCompDirectiveError
-			}
-			containers = append(containers, res...)
-		}
+	containers, err := registry.ContainerEngine().ContainerList(registry.GetContext(), listOpts)
+	if err != nil {
+		cobra.CompErrorln(err.Error())
+		return nil, cobra.ShellCompDirectiveError
 	}
 
 	for _, c := range containers {
-		// include ids in suggestions if more then 2 chars are typed
-		if len(toComplete) > 1 && strings.HasPrefix(c.ID, toComplete) {
+		// include ids in suggestions if cType == completeIDs or
+		// more then 2 chars are typed and cType == completeDefault
+		if ((len(toComplete) > 1 && cType == completeDefault) ||
+			cType == completeIDs) && strings.HasPrefix(c.ID, toComplete) {
```

Comment:
Does Go have the concept of bitmasks? All these `==` and `!=` and `&&` are a little awkward for my eye to read; it seems they would be clearer with `if cType & completeIDs` or `& completeNames`

Reply:
Even if I use bitmasks I still would need to differentiate between completeIDs and completeDefault so it would look exactly the same.
Or I'm missing something?

Reply:
Because of the `need two characters` requirement? Something like this could handle it:
```go
const (
       completeNames = 1
       completeIDs   = 2
       completePartialIDs = 4
 )
...
               if ((len(toComplete) > 1 && cType & completePartialIDs) ||
                       cType & completeIDs) && strings.HasPrefix(c.ID, toComplete) {
```
Unfortunately that's not as readable as I'd expected and I can't quickly think of a good alternative. OK, never mind then.